### PR TITLE
[Feature] show federated username in profile and conversation

### PR DESCRIPTION
**Pull request by David-Henner**
## What's new in this PR?

https://wearezeta.atlassian.net/browse/FS-117

### Changes

Added support for federated usernames in: 
- conversation details
- conversation title
- connection requests
- user profile

Updated logic for the handle display string.

### Tests

Updated snapshot tests. 
Refactored snapshot tests using `FBSnapshotTestCase` into snapshots tests based on `SnapshotTesting` framework.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Availability/TitleView.swift [hunk 1]
```diff
@@ -24,7 +24,9 @@ class TitleView: UIView {
     var titleColor, titleColorSelected: UIColor?
     var titleFont: UIFont?
     let titleButton = UIButton()
+    let subtitleLabel = UILabel()
     var tapHandler: ((UIButton) -> Void)?
+    let stackView = UIStackView(axis: .vertical)
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Availability/TitleView.swift:29]: make this and subtitleLabel private if they are not access by outside

File: Wire-iOS/Sources/UserInterface/ConnectRequests/UserConnectionView.swift [hunk 2]
```diff
@@ -98,8 +98,9 @@ final class UserConnectionView: UIView, Copyable {
     }
 
     private var handleLabelText: NSAttributedString? {
-        guard let handle = user.handle, handle.count > 0 else { return nil }
-        return ("@" + handle) && [
+        guard let handle = user.handleDisplayString(withDomain: user.isFederated), !handle.isEmpty else { return nil }
+
+        return handle && [
             .foregroundColor: UIColor.from(scheme: .textDimmed),
             .font: FontSpec(.small, .semibold).font!
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/ConnectRequests/UserConnectionView.swift:105]:
```suggestion
            .font: UIFont.smallSemiboldFont
```

Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/ConnectRequests/UserConnectionView.swift:105]: remove force unwrap

**Review by billypchan — COMMENTED** [review]
comment

**Review by billypchan — APPROVED** [review]

**Review by typfel — APPROVED** [review]

**Comment by billypchan** [comment]
test failed for 
❌  /Users/runner/work/1/s/Wire-iOS Tests/ConversationTitleViewTests.swift:120:13: 'titleButton' is inaccessible due to 'private' protection level

        sut.titleButton.sendActions(for: .touchUpInside)

may be caused by my refactoring, will update this PR

**Pull request merged**